mlx/ops.cpp

```diff
@@ -1051,6 +1051,10 @@ array array_equal(
   }
 }
 
+array isnan(const array& a, StreamOrDevice s /* = {} */) {
+  return a != a;
```

Comment:
This needs to use the stream, 
```suggestion
  return not_equal(a, a, s);
```